### DISCUSSION TITLE
Draggable Buttons please!!!

### DISCUSSION THREAD

**Discussion by peer-pri** [TomSchimansky/CustomTkinter#2092] [Ideas]
Please make widgets have an attribute called say "draggable" and set it default to False.

When the value of this option is True
-users can drag the widget or at least be able to drag certain creative elements in canvas, or an image.
-allow it to be draggable only inside its parent and not outside.
-add some of your own creative ideas.

**Comment by Akascape** [comment]
@peer-pri You have to create this yourself. Like I have created this node based systes using tkinter canvas: https://github.com/Akascape/TkNodeSystem

You can take the reference how I binded the mouse movement with the drawn rectangle shapes. You can apply the same with ctk widgets but their corners will overlap with each other (it will not be transparent). I recommend you to draw the thing inside the canvas and add mouse movement to them. It is very easy to add mouse movement, there are many other examples available.

![node_menu](https://github.com/Akascape/TkNodeSystem/assets/89206401/0ba5ba42-4787-4b94-8b40-682084df2e80)
